bootcfg/http/ignition.go

```diff
@@ -59,28 +62,70 @@ func ignitionHandler(srv server.Server) ContextHandler {
 			return
 		}
 
-		// Unmarshal YAML or JSON Ignition config
-		var cfg ignition.Config
-		if isYAML(profile.IgnitionId) {
-			if err := yaml.Unmarshal(buf.Bytes(), &cfg); err != nil {
-				log.Errorf("error parsing YAML Ignition config: %v", err)
-				http.NotFound(w, req)
-				return
-			}
-		} else {
-			cfg, err = ignition.Parse(buf.Bytes())
-			if err != nil {
-				log.Errorf("error parsing JSON Ignition config: %v", err)
-				http.NotFound(w, req)
-				return
-			}
+		// Unmarshal YAML or JSON to Ignition V2
+		cfg, err := parseToV2(buf.Bytes())
+		if err == nil {
+			renderJSON(w, cfg)
+			return
+		}
+
+		// Unmarshal YAML or JSON to Ignition V1
+		oldCfg, err := parseToV1(buf.Bytes())
+		if err == nil {
+			renderJSON(w, oldCfg)
+			return
 		}
-		// Marshal Ignition config as JSON HTTP response
-		renderJSON(w, cfg)
+
+		log.Errorf("error parsing Ignition config: %v", err)
+		http.NotFound(w, req)
+		return
 	}
 	return ContextHandlerFunc(fn)
 }
 
-func isYAML(filename string) bool {
-	return strings.HasSuffix(filename, ".yaml") || strings.HasSuffix(filename, ".yml")
+// parseToV2 parses raw JSON in Ignition v2 format and returns the
+// Ignition v2 Config struct.
+func parseToV2(data []byte) (cfg ignitionTypes.Config, err error) {
+	// parse JSON v2 to Ignition
+	cfg, err = ignition.ParseFromLatest(data)
+	if err == nil {
+		return cfg, nil
+	}
+	switch majorVersion(data) {
```

Comment:
Thanks, just a holdover from the auto-upgrade V1 -> V2 flow I was toying with, I'll change it to that.